test/suites/integration/ami_test.go

```diff
@@ -211,6 +215,43 @@ var _ = Describe("AMI", func() {
 
 			env.ExpectInstance(pod.Spec.NodeName).To(HaveField("ImageId", HaveValue(Equal(customAMI))))
 		})
+		It("should have the AWSNodeTemplateStatus for AMIs using wildcard", func() {
+			provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+				AWS: v1alpha1.AWS{
+					SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+					SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+				},
+				AMISelector: map[string]string{"aws::name": "*"},
+			})
+
+			env.ExpectCreated(provider)
+			EventuallyExpectAMIsToExist(provider)
+		})
+		It("should have the AWSNodeTemplateStatus for AMIs using tags", func() {
+			provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+				AWS: v1alpha1.AWS{
+					SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+					SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+				},
+				AMISelector: map[string]string{"aws-ids": customAMI},
+			})
+
+			env.ExpectCreated(provider)
+			EventuallyExpectAMIsToExist(provider)
+
+			var ant v1alpha1.AWSNodeTemplate
+			Expect(env.Client.Get(env, client.ObjectKeyFromObject(provider), &ant)).To(Succeed())
+			amiIDsInStatus := lo.Map(ant.Status.AMIs, func(ami v1alpha1.AMI, _ int) string {
+				return ami.ID
+			})
+			Expect(amiIDsInStatus).To(Equal([]string{customAMI}))
```

Comment:
nit: If there's just one AMI that matches, could you consider asserting the length of the AMI and then just ensuring that the first element's ID equals the ID that you would expect